### PR TITLE
storagefsm: Fix expired ticket retry loop

### LINKED ISSUE #4865: PC tickets expire, seems new ticket is already expired at retrieval time
All my sectors during the weekend got stuck in PC2 due to some new GPU scheduling bug in 1.1.3. I've reconfigured things to not have concurrent GPU handling, after which PC2 seems to have worked, but all the sectors started looping between PC2->PC1->PC2, etc. First two time it said the sector ticket expired. Two bugs though:

- If the ticket expired, why on earth try it again with the same expired ticket. Surely we don't expect it to go back in time and finish it the second time before the deadline.
- After 2 failures, the logs seem to show that a new ticket was requested, which was already expired upon request:

```
Event Log:
0.	2020-11-14 12:28:35 +0200 EET:	[event;sealing.SectorStartCC]	{"User":{"ID":569,"SectorType":3,"Pieces":[{"Piece":{"Size":34359738368,"PieceCID":{"/":"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq"}},"DealInfo":null}]}}
1.	2020-11-14 12:28:35 +0200 EET:	[event;sealing.SectorPacked]	{"User":{"FillerPieces":null}}
2.	2020-11-14 12:28:37 +0200 EET:	[event;sealing.SectorTicket]	{"User":{"TicketValue":"M1t/hKgo2wxZEn6AjiYgtgotBAsf0yzmxuY5J4q4MnE=","TicketEpoch":233877}}
3.	2020-11-14 14:10:30 +0200 EET:	[event;sealing.SectorRestart]	{"User":{}}
4.	2020-11-14 14:10:31 +0200 EET:	[event;sealing.SectorSealPreCommit1Failed]	{"User":{}}
	seal pre commit(1) failed: worker restarted
5.	2020-11-14 14:11:31 +0200 EET:	[event;sealing.SectorRetrySealPreCommit1]	{"User":{}}
6.	2020-11-14 14:12:50 +0200 EET:	[event;sealing.SectorRestart]	{"User":{}}
7.	2020-11-14 14:23:56 +0200 EET:	[event;sealing.SectorRestart]	{"User":{}}
8.	2020-11-14 14:23:57 +0200 EET:	[event;sealing.SectorSealPreCommit1Failed]	{"User":{}}
	seal pre commit(1) failed: worker restarted
9.	2020-11-14 14:24:57 +0200 EET:	[event;sealing.SectorRetrySealPreCommit1]	{"User":{}}
10.	2020-11-14 14:25:07 +0200 EET:	[event;sealing.SectorSealPreCommit1Failed]	{"User":{}}
	seal pre commit(1) failed: acquiring sector paths: local acquire error: finding best storage for allocating : no good path found
11.	2020-11-14 14:26:07 +0200 EET:	[event;sealing.SectorRetrySealPreCommit1]	{"User":{}}
12.	2020-11-14 21:59:30 +0200 EET:	[event;sealing.SectorSealPreCommit1Failed]	{"User":{}}
	seal pre commit(1) failed: acquiring sector paths: reserving storage space: can't reserve 484472310988 bytes in '/mnt/worker-hddb' (id:4d8739f3-b6f0-48a3-aca6-965fc7fdced5), only 43306869556 available
13.	2020-11-14 22:00:30 +0200 EET:	[event;sealing.SectorRetrySealPreCommit1]	{"User":{}}
14.	2020-11-15 03:00:49 +0200 EET:	[event;sealing.SectorPreCommit1]	{"User":{"PreCommit1Out":"eyJyZWdpc3RlcmVkX3Byb29mIjoiU3RhY2tlZERyZzMyR2lCVjEiLCJsYWJlbHMiOnsiU3RhY2tlZERyZzMyR2lCVjEiOnsibGFiZWxzIjpbeyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci0xIiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvbW50L3dvcmtlci1oZGRhL2NhY2hlL3MtdDAxOTEwMC01NjkiLCJpZCI6ImxheWVyLTIiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9tbnQvd29ya2VyLWhkZGEvY2FjaGUvcy10MDE5MTAwLTU2OSIsImlkIjoibGF5ZXItMyIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci00Iiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvbW50L3dvcmtlci1oZGRhL2NhY2hlL3MtdDAxOTEwMC01NjkiLCJpZCI6ImxheWVyLTUiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9tbnQvd29ya2VyLWhkZGEvY2FjaGUvcy10MDE5MTAwLTU2OSIsImlkIjoibGF5ZXItNiIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci03Iiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvbW50L3dvcmtlci1oZGRhL2NhY2hlL3MtdDAxOTEwMC01NjkiLCJpZCI6ImxheWVyLTgiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9tbnQvd29ya2VyLWhkZGEvY2FjaGUvcy10MDE5MTAwLTU2OSIsImlkIjoibGF5ZXItOSIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci0xMCIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci0xMSIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N31dLCJfaCI6bnVsbH19LCJjb25maWciOnsicGF0aCI6Ii9tbnQvd29ya2VyLWhkZGEvY2FjaGUvcy10MDE5MTAwLTU2OSIsImlkIjoidHJlZS1kIiwic2l6ZSI6MjE0NzQ4MzY0Nywicm93c190b19kaXNjYXJkIjo3fSwiY29tbV9kIjpbNywxMjYsOTUsMjIyLDUzLDE5NywxMCwxNDcsMywxNjUsODAsOSwyMjcsNzMsMTM4LDc4LDE5MCwyMjMsMjQzLDE1Niw2NiwxODMsMTYsMTgzLDQ4LDIxNiwyMzYsMTIyLDE5OSwxNzUsMTY2LDYyXX0="}}
15.	2020-11-15 20:04:00 +0200 EET:	[event;sealing.SectorRestart]	{"User":{}}
16.	2020-11-15 20:59:47 +0200 EET:	[event;sealing.SectorPreCommit2]	{"User":{"Sealed":{"/":"bagboea4b5abcbv6obb4fqbx2gammxd4g2agzjyjkoo3bcgdlmtf3ywlo2vl4ucsu"},"Unsealed":{"/":"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq"}}}
17.	2020-11-15 20:59:47 +0200 EET:	[event;sealing.SectorSealPreCommit1Failed]	{"User":{}}
	ticket expired: ticket expired: seal height: 234777, head: 238679
18.	2020-11-15 21:00:47 +0200 EET:	[event;sealing.SectorRetrySealPreCommit1]	{"User":{}}
19.	2020-11-16 03:02:54 +0200 EET:	[event;sealing.SectorPreCommit1]	{"User":{"PreCommit1Out":"eyJyZWdpc3RlcmVkX3Byb29mIjoiU3RhY2tlZERyZzMyR2lCVjEiLCJsYWJlbHMiOnsiU3RhY2tlZERyZzMyR2lCVjEiOnsibGFiZWxzIjpbeyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci0xIiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvbW50L3dvcmtlci1oZGRhL2NhY2hlL3MtdDAxOTEwMC01NjkiLCJpZCI6ImxheWVyLTIiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9tbnQvd29ya2VyLWhkZGEvY2FjaGUvcy10MDE5MTAwLTU2OSIsImlkIjoibGF5ZXItMyIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci00Iiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvbW50L3dvcmtlci1oZGRhL2NhY2hlL3MtdDAxOTEwMC01NjkiLCJpZCI6ImxheWVyLTUiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9tbnQvd29ya2VyLWhkZGEvY2FjaGUvcy10MDE5MTAwLTU2OSIsImlkIjoibGF5ZXItNiIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci03Iiwic2l6ZSI6MTA3Mzc0MTgyNCwicm93c190b19kaXNjYXJkIjo3fSx7InBhdGgiOiIvbW50L3dvcmtlci1oZGRhL2NhY2hlL3MtdDAxOTEwMC01NjkiLCJpZCI6ImxheWVyLTgiLCJzaXplIjoxMDczNzQxODI0LCJyb3dzX3RvX2Rpc2NhcmQiOjd9LHsicGF0aCI6Ii9tbnQvd29ya2VyLWhkZGEvY2FjaGUvcy10MDE5MTAwLTU2OSIsImlkIjoibGF5ZXItOSIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci0xMCIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N30seyJwYXRoIjoiL21udC93b3JrZXItaGRkYS9jYWNoZS9zLXQwMTkxMDAtNTY5IiwiaWQiOiJsYXllci0xMSIsInNpemUiOjEwNzM3NDE4MjQsInJvd3NfdG9fZGlzY2FyZCI6N31dLCJfaCI6bnVsbH19LCJjb25maWciOnsicGF0aCI6Ii9tbnQvd29ya2VyLWhkZGEvY2FjaGUvcy10MDE5MTAwLTU2OSIsImlkIjoidHJlZS1kIiwic2l6ZSI6MjE0NzQ4MzY0Nywicm93c190b19kaXNjYXJkIjo3fSwiY29tbV9kIjpbNywxMjYsOTUsMjIyLDUzLDE5NywxMCwxNDcsMywxNjUsODAsOSwyMjcsNzMsMTM4LDc4LDE5MCwyMjMsMjQzLDE1Niw2NiwxODMsMTYsMTgzLDQ4LDIxNiwyMzYsMTIyLDE5OSwxNzUsMTY2LDYyXX0="}}
20.	2020-11-16 04:38:42 +0200 EET:	[event;sealing.SectorPreCommit2]	{"User":{"Sealed":{"/":"bagboea4b5abcbv6obb4fqbx2gammxd4g2agzjyjkoo3bcgdlmtf3ywlo2vl4ucsu"},"Unsealed":{"/":"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq"}}}
21.	2020-11-16 04:38:42 +0200 EET:	[event;sealing.SectorSealPreCommit1Failed]	{"User":{}}
	ticket expired: ticket expired: seal height: 234777, head: 239597
22.	2020-11-16 04:39:42 +0200 EET:	[event;sealing.SectorRetrySealPreCommit1]	{"User":{}}
23.	2020-11-16 07:34:49 +0200 EET:	[event;sealing.SectorRestart]	{"User":{}}
24.	2020-11-16 07:34:49 +0200 EET:	[event;sealing.SectorOldTicket]	{"User":{}}
25.	2020-11-16 07:34:50 +0200 EET:	[event;sealing.SectorTicket]	{"User":{"TicketValue":"zM0j4kiTXVjVETOpa1VqI6w2q057Akra0QkX9lVzIfU=","TicketEpoch":239049}}
```

### PR DESCRIPTION
**Pull request by magik6k**
Fixes  #4865

### CODE REVIEW

**Review by arajasek — APPROVED** [review]

**Comment by magik6k** [comment]
(broke something in windowpost test)

**Comment by s0nik42** [comment]
Is this error related to the bug : 

2020-11-18T23:52:09.020+0100	WARN	sectors	storage-sealing/fsm.go:475	sector 1405 got error event sealing.SectorSealPreCommit1Failed: getting ticket failed: getting precommit info: sectorNumber is allocated, but PreCommit info wasn't found on chain

**Comment by dodohack** [comment]
I encountered similar bug as well, here is the log.

```
2020-11-21T10:01:36.379Z	WARN	sectors	storage-sealing/fsm.go:507	sector 19 got error event sealing.SectorSealPreCommit1Failed: ticket expired: ticket expired: seal height: 249141, head: 254883
2020-11-21T10:01:36.380Z	INFO	sectors	storage-sealing/states_failed.go:26	SealPreCommit1Failed(19), waiting 59.619750759s before retrying
2020-11-21T10:02:36.014Z	DEBUG	advmgr	sector-storage/sched.go:354	SCHED 1 queued; 10 open windows
```

Basically, the sector 19 had finished PC1 and was running PC2 for more than 20 mins, but when this error "ticket expired" message  happened, it caused sector 19 from running PC2, back to re-run PC1 instead.

My lotus/miner/worker version is: version 1.2.1+git.df66f48f6

**Comment by s0nik42** [comment]
@magik6k is it going to be corrected on the next release or should be remove the sectors manually ? my miners is saturated of them.
Thanks

**Comment by Shekelme** [comment]
`2020-11-21T16:54:15.076+0200    WARN    sectors storage-sealing/fsm.go:507      sector 4 got error event sealing.SectorSealPreCommit1Failed: ticket expired: ticket expired: seal height: 250201, head: 255468 `

What should I do with this sector in endless sealing loop?

File: extern/storage-sealing/states_sealing.go [hunk 1]
```diff
@@ -297,8 +294,10 @@ func (m *Sealing) handlePreCommitWait(ctx statemachine.Context, sector SectorInf
 	switch mw.Receipt.ExitCode {
 	case exitcode.Ok:
 		// this is what we expect
+	case exitcode.SysErrInsufficientFunds:
+		fallthrough
 	case exitcode.SysErrOutOfGas:
-		// gas estimator guessed a wrong number
+		// gas estimator guessed a wrong number / out of funds:
```
Comment (Kubuxu) [inline extern/storage-sealing/states_sealing.go:300]: Good to see this, as at the upgrade epoch we might hit it.

**Review by Kubuxu — APPROVED** [review]

**Pull request merged**